packages/pg/lib/connection.js

```diff
@@ -178,22 +198,30 @@ class Connection extends EventEmitter {
   }
 
   ref() {
-    this.stream.ref()
+    this.stream.socket._socket.ref()
   }
 
   unref() {
-    this.stream.unref()
+    this.stream.socket._socket.unref()
   }
 
   end() {
     // 0x58 = 'X'
     this._ending = true
     if (!this._connecting || !this.stream.writable) {
-      this.stream.end()
+      if(this.stream.socket) {
+        this.stream.socket.close()
+      } else {
+        this.stream.end()
+      }
       return
     }
     return this.stream.write(endBuffer, () => {
-      this.stream.end()
+      if(this.stream.socket) {
+        this.stream.socket.close()
+      } else {
+        this.stream.end()
```

Comment:
When does each occur?

Reply:
This is purely for unit/integration test purposes -- websockets have a stream.socket.close() function to end them, while the memoryStream sockets used in testing have a stream.end() to close.